lib/src/utils/network_manager.dart

```diff
@@ -3,22 +3,37 @@ import 'dart:async';
 import 'package:instabug_flutter/instabug_flutter.dart';
 
 typedef ObfuscateLogCallback = FutureOr<NetworkData> Function(NetworkData data);
+typedef OmitLogCallback = FutureOr<bool> Function(NetworkData data);
 
 /// Mockable [NetworkManager] responsible for processing network logs
 /// before they are sent to the native SDKs.
 class NetworkManager {
   ObfuscateLogCallback? _obfuscateLogCallback;
+  OmitLogCallback? _omitLogCallback;
 
   // ignore: use_setters_to_change_properties
   void setObfuscateLogCallback(ObfuscateLogCallback callback) {
     _obfuscateLogCallback = callback;
   }
 
+  // ignore: use_setters_to_change_properties
+  void setOmitLogCallback(OmitLogCallback callback) {
+    _omitLogCallback = callback;
+  }
+
   FutureOr<NetworkData> obfuscateLog(NetworkData data) {
     if (_obfuscateLogCallback == null) {
       return data;
     }
 
     return _obfuscateLogCallback!(data);
   }
+
+  FutureOr<bool> omitLog(NetworkData data) {
```

Comment:
How about we add some docs to this API:
```dart
/// Registers a callback to omit some of the network log data.
///
/// Use this method to set a callback that decides whether to omit
/// specific network log data before recording or displaying it.
///
/// The [callback] function takes a [NetworkData] object and should
/// return a boolean value indicating whether to omit (true) or include
/// (false) the data.
///
/// Example:
/// NetworkLogger. omitLog((data) {
///   // Implement logic to decide if the data should be omitted. for instance check if it contains a specific keyword return false otherwise true
///   return someCondition ? true : false;
/// });
FutureOr<bool> omitLog(NetworkData data) {
```

Reply:
@a7medev, I've made the updated mentioned here with a slight adjustment, could you please check them?